parser_library/src/expressions/conditional_assembly/ca_operator_binary.cpp

```diff
@@ -287,31 +290,76 @@ context::SET_t ca_function_binary_operator::operation(
     return context::SET_t(expr_kind);
 }
 
-int ca_function_binary_operator::compare_string(const context::C_t& lhs, const context::C_t& rhs)
+std::strong_ordering ca_function_binary_operator::compare_string(
+    const context::C_t& lhs, const context::C_t& rhs) noexcept
 {
-    int diff = (int)lhs.size() - (int)rhs.size();
+    bool left_smaller = false;
+    bool right_smaller = false;
 
-    if (diff != 0)
-        return diff;
+    const auto* l = std::to_address(lhs.cbegin());
+    const auto* r = std::to_address(rhs.cbegin());
+    const auto* const le = std::to_address(lhs.cend());
+    const auto* const re = std::to_address(rhs.cend());
 
-    return ebcdic_encoding::to_ebcdic(lhs).compare(ebcdic_encoding::to_ebcdic(rhs));
+    while (l != le && r != re)
+    {
+        const unsigned char lc = ebcdic_encoding::to_ebcdic(l);
```

Comment:
I get that `to_ebcdic()` increases the value of `l`, however, I was confused twice in this PR already, why the loop seemingly doesn't have an end condition...

How do you feel about `to_ebcdic()` returning something like a `std::pair<const unsigned char, size_t>`. The `second` parameter of the `pair` could indicate how many bytes were processed

Then the code could look like:

```
auto res = ebcdic_encoding::to_ebcdic(l);
const unsigned char lc = res.first;
l += res.second;
```
and the end condition would be clearer

Reply:
I understand the objection, but at least in MSVC this would cause the return value be passed via stack, I'd like to keep this function as lean as possible.

Reply:
But I'll think about it more...